### PR TITLE
feat(AppContainerModel): expose bindable var to allow custom init mes…

### PR DESCRIPTION
**Pull request by jacob-xhio**
…sages for hoist apps

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by jacob-xhio** [comment]
Screenshot 2024-08-22 at 11 03 09 AM

File: appcontainer/AppContainerModel.ts [hunk 1]
```diff
@@ -72,10 +72,16 @@ export class AppContainerModel extends HoistModel {
     appSpec: AppSpec = null;
     appModel: HoistAppModel = null;
 
+    //---------------------------------
+    // Configurable Application State
+    //--------------------------------
+    @bindable initializingLoadMaskMessage;
```
Comment (amcclain) [inline appcontainer/AppContainerModel.ts:78]: Would add the type here - `ReactNode`

File: appcontainer/AppContainerModel.ts [hunk 2]
```diff
@@ -316,6 +322,10 @@ export class AppContainerModel extends HoistModel {
         return !isEmpty(this.aboutDialogModel.getItems());
     }
 
+    setInitializingLoadMaskMessage(message: string) {
+        this.initializingLoadMaskMessage = message;
+    }
```
Comment (amcclain) [inline appcontainer/AppContainerModel.ts:327]: Sorry as I think I had this in the spec, but realized that we've recently trended more towards exposing `@bindable` properties directly for use by callers - it's reasonable for apps to set them directly, and we can remove the setter here.  That tracks more with some other updates done recently.

To emphasize though that the property is a public API, please add a proper `/** */` doc string, in addition to the type.  Can replace that banner-style block comment

**Review by amcclain — COMMENTED** [review]
Great thanks - small comments below, also pls add the same to mobile

File: appcontainer/AppContainerModel.ts (hunk 2 above)
Comment (jacob-xhio) [reply appcontainer/AppContainerModel.ts]: Resolved, added docstring explaining the purpose of the bindable.

**Pull request merged**